ledger/sandbox/src/main/scala/com/digitalasset/ledger/server/LedgerApiServer/LedgerApiServer.scala

```diff
@@ -46,34 +33,33 @@ object LedgerApiServer {
       config: SandboxConfig,
       //even though the port is in the config as well, in case of a reset we have to keep the port to what it was originally set for the first time
       serverPort: Int,
-      optTimeServiceBackend: Option[TimeServiceBackend],
-      optResetService: Option[SandboxResetService])(
+      timeServiceBackend: Option[TimeServiceBackend],
+      resetService: Option[SandboxResetService])(
       implicit mat: ActorMaterializer): LedgerApiServer = {
 
     new LedgerApiServer(
-      ledgerBackend,
-      timeProvider,
-      engine,
+      (am: ActorMaterializer, esf: ExecutionSequencerFactory) =>
+        ApiServices.create(config, ledgerBackend, engine, timeProvider, timeServiceBackend)(
+          am,
+          esf),
       config,
       serverPort,
-      optTimeServiceBackend,
-      optResetService,
+      timeServiceBackend,
+      resetService,
       config.address,
       config.tlsConfig.flatMap(_.server)
-    ).start()
+    )
   }
 }
 
 class LedgerApiServer(
-    ledgerBackend: LedgerBackend,
-    timeProvider: TimeProvider,
-    engine: Engine,
+    createApiServices: (ActorMaterializer, ExecutionSequencerFactory) => ApiServices,
     config: SandboxConfig,
     serverPort: Int,
-    optTimeServiceBackend: Option[TimeServiceBackend],
-    optResetService: Option[SandboxResetService],
-    addressOption: Option[String],
-    maybeBundle: Option[SslContext] = None)(implicit mat: ActorMaterializer)
+    timeServiceBackend: Option[TimeServiceBackend],
```

Comment:
`timeServiceBackend` is not used anywhere in this class and probably should be removed or folded into the `ApiServices` like the `resetService`.

Reply:
yes, I have a follow-up pr coming take care of that. I wanted to do the refactoring in small steps, and was hoping this can land early Friday.

Reply:
👍